modules/avcodec/encode.c

```diff
@@ -205,8 +210,16 @@ static int open_encoder(struct videnc_state *st,
 	av_opt_set_defaults(st->ctx);
 
 	st->ctx->bit_rate  = prm->bitrate;
-	st->ctx->width     = size->w;
-	st->ctx->height    = size->h;
+
+	if (sz_dst.w <= 0) {;
+		sz_dst.w = size->w;
+	}
+	if (sz_dst.h <= 0) {
+		sz_dst.h = size->h;
+	}
+
+	st->ctx->width     = sz_dst.w;
+	st->ctx->height    = sz_dst.h;
```

Comment:
not sure if this is needed. Each input frame to the encoder contains the size of the image,
and the size might also change in a stream.